src/main/java/uk/gov/companieshouse/presenter/account/model/PresenterAccountAddress.java

```diff
@@ -0,0 +1,206 @@
+package uk.gov.companieshouse.presenter.account.model;
+
+import java.util.Optional;
+
+import org.springframework.data.mongodb.core.mapping.Field;
+
+import uk.gov.companieshouse.presenter.account.exceptionhandler.ValidationException;
+import uk.gov.companieshouse.presenter.account.model.request.presenter.account.PresenterAddress;
+import uk.gov.companieshouse.presenter.account.validation.utils.StringValidator;
+
+public class PresenterAccountAddress {
+
+    private static final int PREMISES_MAX_LENGTH = 40;
+    private static final int LINE_1_MAX_LENGTH = 40;
+    private static final int LINE_2_MAX_LENGTH = 40;
+    private static final int COUNTY_MAX_LENGTH = 40;
+    private static final int COUNTRY_MAX_LENGTH = 40;
+    private static final int POSTCODE_MAX_LENGTH = 10;
+    
+    @Field
+    private final String premises;
+
+    @Field
+    private final String addressLine1;
+
+    @Field
+    private final String addressLine2;
+
+    @Field
+    private final String county;
+
+    @Field
+    private final String country;
+
+    @Field("postCode")
+    private final String postcode;
+
+    public static PresenterAccountAddress createPresenterAccountAddress(PresenterAddress presenterAddress){
+        if (presenterAddress != null) {
+            return new PresenterAccountAddress(presenterAddress.premises(), 
+            presenterAddress.addressLine1(), 
+            presenterAddress.addressLine2(), 
+            presenterAddress.county(), 
+            presenterAddress.country(), 
+            presenterAddress.postcode());
+        }
+        else{
+            throw new ValidationException("Presenter address info is missing");
+        }
+    }
+
+    public PresenterAccountAddress(String premises, String addressLine1, String addressLine2, String county, String country,
+            String postcode) {
+        this.premises = validatePremises(premises);
+        this.addressLine1 = validateAddressLine1(addressLine1);
+        this.addressLine2 = validateAddressLine2(addressLine2);
+        this.county = validatedCounty(county);
+        this.country = validateCountry(country);
+        this.postcode = validatePostcode(postcode);
+    }
+
+    private String validatePremises(String premises){
+        final int maxLength = PREMISES_MAX_LENGTH;
+        final String exceptionMessage = "premises failed validation";
+        return getValidatedLine(premises, maxLength, exceptionMessage);
```

Comment:
I personally think this looks better
```suggestion
        return getValidatedLine(premises, maxLength, "premises failed validation");
```